overlord/assertstate/assertstate.go

```diff
@@ -352,11 +356,17 @@ func AutoRefreshAssertions(s *state.State, userID int) error {
 }
 
 // RefreshSnapAssertions tries to refresh all snap-centered assertions
-func RefreshSnapAssertions(s *state.State, userID int) error {
-	opts := &RefreshAssertionsOptions{IsAutoRefresh: false}
+func RefreshSnapAssertions(s *state.State, userID int, opts *RefreshAssertionsOptions) error {
```

Comment:
don't we need to do something also in AutoRefreshAssertions setting IsRefreshOfAllSnaps to true or letting it be passed in?

Reply:
Hmm I think you're right, due to refresh control with gate-auto-refresh hooks.
I think there is a problem with that though as we refresh assertions very early in AutoRefresh(), so maybe for refresh control we need to delay this until we know the outcome of the hooks and what's going to be really refreshed; that issue is probably for a separate PR, but I'll add opts to AutoRefreshAssertions as well. Thanks

Reply:
Udated

Reply:
Reverted my changes to AutoRefreshAssertions as discussed on the standup. We don't need to do anything about it in AutoRefreshAssertions because IsRefreshOfAllSnaps flag only affects RefreshSnapAssertions()logic (we refresh validation sets or not based on it, but in AutoRefreshAssertions it's unconditional).